README.md

```diff
@@ -12,15 +12,36 @@
-* normalized name - name that is in normalized form according to the ENS Normalization Standard. This means `name == ens_normalize(name)`.
-* normalizable name - name that is normalized or that can be converted into a normalized name using `ens_normalize`.
-* beautiful name - name that is normalizable and is equal to itself when using `ens_beautify`. This means `name == ens_beautify(name)`.
-* disallowed name - name that is not normalizable. This means `ens_normalize(name)` raises a `DisallowedNameError`.
-* curable name - name that is normalizable, or a name in the subset of disallowed names that can still be converted into a normalized name using `ens_cure`.
-* fatal error - a `DisallowedNameError` object thrown by `ens_normalize` that contains only general information about the error and no suggested fixes.
-* curable error - a `CurableError` object (inherits from `DisallowedNameError`) thrown by `ens_normalize` that contains information about a suggested potential fix for the error.
+**Foundations**
+* **sequence** - a Unicode string containing any number of characters.
+* **label** - a sequence of any length (including 0) that does not contain a `.` (full stop / period) character, e.g.  `abc` or `eth`.
+* **name** - a series of any number of labels (including 0) separated by `.` (full stop / period) characters, e.g. `abc.eth`.
+
+**Names**
+* **normalized name** - name that is in normalized form according to the ENS Normalization Standard. This means `name == ens_normalize(name)`. A normalized name always contains at least 1 label. All labels in a normalized name always contain a sequence of at least 1 character.
+* **normalizable name** - name that is normalized or that can be converted into a normalized name using `ens_normalize`.
+* **beautiful name** - name that is normalizable and is equal to itself when using `ens_beautify`. This means `name == ens_beautify(name)`. For all normalizable names `ens_normalize(ens_beautify(name)) == ens_normalize(name)`.
+* **disallowed name** - name that is not normalizable. This means `ens_normalize(name)` raises a `DisallowedSequence`.
+* **curable name** - name that is normalizable, or a name in the subset of disallowed names that can still be converted into a normalized name using `ens_cure`.
+* **empty name** - name that is the empty string. An empty name is disallowed and not curable.
+* **namehashable name** - name that is ready for for use with the ENS `namehash` function. Only normalized and empty names are namehashable.
+
+**Sequences**
+* **unnormalized sequence** - a sequence from a name that is not in normalized form according to the ENS Normalization Standard.
+* **normalization suggestion** - a sequence suggested as an in-place replacement for an unnormalized sequence.
+* **normalizable sequence** - an unnormalized sequence containing a normalization suggestion that is automatically applied using `ens_normalize` and `ens_cure`.
```

Comment:
Note that I changed "or" to "and".
"or" made it seem like every curable sequence is also normalizable.

Reply:
I don't see it. `or` and `and` is ok for me. 
However, "sequence containing sequence" is a little confusing, like it is a subsequence - but maybe it my poor English.